scrapli/transport/plugins/telnet/transport.py

```diff
@@ -119,21 +148,11 @@ def _handle_control_chars(self) -> None:
 
         control_buf = b""
 
-        original_socket_timeout = self._base_transport_args.timeout_socket
-        self._set_socket_timeout(self._base_transport_args.timeout_socket / 4)
+        while self._raw_buf:
+            c, self._raw_buf = self._raw_buf[:1], self._raw_buf[1:]
+            if not c:
+                raise ScrapliConnectionNotOpened("server returned EOF, connection not opened")
 
-        while True:
-            try:
-                c = self._read(1)
-                if not c:
-                    raise ScrapliConnectionNotOpened("server returned EOF, connection not opened")
-            except TimeoutError:
```

Comment:
since we aren't reading from the socket now as we just read from the buf I removed the try/except here since we shouldn't encounter any exception I think.